packages/next-swc/crates/next-core/js/src/entry/app/route.ts

```diff
@@ -1,39 +1,74 @@
 // IPC need to be the first import to allow it to catch errors happening during
 // the other imports
 import startHandler from '../../internal/api-server-handler'
-import { runEdgeFunction } from '../../internal/edge'
 
-import { join } from 'path'
-
-import 'next/dist/server/node-polyfill-fetch.js'
-
-import chunkGroup from 'ROUTE_CHUNK_GROUP'
+import '../../polyfill/app-polyfills'
 
 import {
   NodeNextRequest,
   NodeNextResponse,
 } from 'next/dist/server/base-http/node'
+import { sendResponse } from 'next/dist/server/send-response'
+import { NextRequestAdapter } from 'next/dist/server/web/spec-extension/adapters/next-request'
+import { RouteHandlerManagerContext } from 'next/dist/server/future/route-handler-managers/route-handler-manager'
+import { parse } from 'url'
+
+import RouteModule from 'ROUTE_MODULE'
+import * as userland from 'ENTRY'
+import { PAGE, PATHNAME } from 'BOOTSTRAP_CONFIG'
+import { BaseNextRequest } from 'next/dist/server/base-http'
+import {
+  addRequestMeta,
+  NextUrlWithParsedQuery,
+} from 'next/dist/server/request-meta'
+import { TLSSocket } from 'tls'
+import { getCloneableBody } from 'next/dist/server/body-streams'
+
+const routeModule = new RouteModule({
+  userland,
+  pathname: PATHNAME,
+  resolvedPagePath: `app/${PAGE}`,
+  nextConfigOutput: undefined,
+})
 
 startHandler(async ({ request, response, query, params, path }) => {
-  const edgeInfo = {
-    name: 'edge',
-    paths: chunkGroup.map((chunk) =>
-      join(process.cwd(), '.next/server/app', chunk)
-    ),
-    wasm: [],
-    env: Object.keys(process.env),
-    assets: [],
-  }
-  await runEdgeFunction({
-    edgeInfo,
-    outputDir: 'app',
-    req: new NodeNextRequest(request),
-    res: new NodeNextResponse(response),
-    query,
+  const req = new NodeNextRequest(request)
+  const res = new NodeNextResponse(response)
+
+  const parsedUrl = parse(req.url!, true)
+  attachRequestMeta(req, parsedUrl, 'localhost', 3000)
```

Comment:
3000 is the port? That's not always 3000. Try `process.env.PORT`

Reply:
I wasn't sure if there is a correct port, there are 3
- the devserver port
- the node.js worker tcp port (`process.env.PORT`)
- the port for the webserver created in `startHandler` (which the request here is on)

Reply:
used the port from headers now, hope that works in all cases